packages/react-components/react-dialog/src/components/DialogTitle/useDialogTitle.tsx

```diff
@@ -15,25 +17,34 @@ import { Dismiss24Regular } from '@fluentui/react-icons';
  * @param ref - reference to root HTMLElement of DialogTitle
  */
 export const useDialogTitle_unstable = (props: DialogTitleProps, ref: React.Ref<HTMLElement>): DialogTitleState => {
-  const { as = 'div', closeButton } = props;
+  const { as, action } = props;
   const modalType = useDialogContext_unstable(ctx => ctx.modalType);
+  const internalStyles = useDialogTitleInternalStyles();
 
   return {
     components: {
       root: 'div',
-      closeButton: 'button',
+      action: 'div',
     },
-    root: getNativeElementProps(as, {
+    root: getNativeElementProps(as ?? 'div', {
       ref,
       id: useDialogContext_unstable(ctx => ctx.dialogTitleID),
       ...props,
     }),
-    closeButton: useARIAButtonShorthand(closeButton, {
+    action: resolveShorthand(action, {
       required: modalType === 'non-modal',
       defaultProps: {
-        type: 'button', // This is added because the default for type is 'submit'
-        'aria-label': 'close', // TODO: find a better way to add internal labels
-        children: <Dismiss24Regular />,
+        children: (
+          <DialogTrigger action="close">
```

Comment:
one problem I can think of is that there will be no easy way for users to override the styles of the button, since it is no longer the slot.

In order to support generic actions, I think it's a good sacrifice... I can't think of a better way 🤷

Reply:
I do believe this is definitely not a big deal, since it's so easy to replace the default action